modules/compiler/spirv-ll/include/spirv-ll/builder.h

```diff
@@ -335,13 +335,32 @@ class Builder {
   /// @return pointer to `DIType` derived from the given `Type`
   llvm::DIType *getDIType(llvm::Type *type);
 
+  /// @brief Gets (or creates) a DIFile for the given OpLine.
+  llvm::DIFile *getOrCreateDIFile(const OpLine *op_line);
+
+  /// @brief Gets (or creates) a DICompileUnit for the given OpLine.
+  llvm::DICompileUnit *getOrCreateDICompileUnit(const OpLine *op_line);
+
+  /// @brief Gets (or creates) a DISubprogram for the given function and
+  /// OpLine.
+  llvm::DISubprogram *getOrCreateDebugFunctionScope(llvm::Function *function,
+                                                    const OpLine *op_line);
+
+  /// @brief Creates a new open line range for the given OpLine, contained
+  /// within the basic block.
```

Comment:
This name "open line range" is used in various places. Is this based on the existing `OpLineRange`? I don't think `Op` means "open" there, I think it's just the range of `OpLine` where `Op` means "operation", and I don't think we use open ranges, which is an established term meaning ranges where the end points are included, as in the open range (1, 10) means the range of `x` such that `x > 1 && x < 10`.

Reply:
You're right about "Op", and no it's not to do with `OpLine` or `OpLineRange`.

I was meaning "open" to mean that it signifies a range that is  "currently open". That is, a range begun by `OpLine` that has not yet been closed by a `OpNoLine` or a block termination instruction.

I'll need to rethink the naming scheme. I probably got a bit lost during the refactoring.

Reply:
Ah, thanks for the explanation, that makes sense. I don't have a better term to suggest right away either, sorry; if one comes to mind before you can update this I'll be sure to post it.

Reply:
It could maybe be a `LineRangeBeginTy` as it's clearly only the beginning of a range, and the type itself doesn't "open" anything - it's other methods that do the opening. We could keep references to "open" in comments when it concerns `getCurrentOpLineRange` and can be clarified (or, you know, just use the word "current").

What do you think?